SignalMessaging/environment/OWSSounds.m

```diff
@@ -207,6 +212,34 @@ + (nullable NSURL *)soundURLForSound:(OWSSound)sound quiet:(BOOL)quiet
     return url;
 }
 
++ (SystemSoundID)systemSoundIDForSound:(OWSSound)sound quiet:(BOOL)quiet
+{
+    return [self.sharedManager systemSoundIDForSound:(OWSSound)sound quiet:quiet];
+}
+
+- (SystemSoundID)systemSoundIDForSound:(OWSSound)sound quiet:(BOOL)quiet
+{
+    NSString *cacheKey = [NSString stringWithFormat:@"%lu:%d", (unsigned long)sound, quiet];
+    NSNumber *_Nullable cachedSoundId = (NSNumber * _Nullable)[self.cachedSoundIDs getWithKey:cacheKey];
+
+    if (cachedSoundId) {
+        OWSAssert([cachedSoundId isKindOfClass:[NSNumber class]]);
+        return (SystemSoundID)cachedSoundId.intValue;
+    }
+
+    NSURL *soundURL = [self.class soundURLForSound:sound quiet:quiet];
+
+    DDLogVerbose(@"%@ creating system sound for %@", self.logTag, soundURL.lastPathComponent);
+    SystemSoundID newSoundID;
+    OSStatus status = AudioServicesCreateSystemSoundID((__bridge CFURLRef _Nonnull)(soundURL), &newSoundID);
+    OWSAssert(status == 0);
+    OWSAssert(newSoundID);
+
+    [self.cachedSoundIDs setWithKey:cacheKey value:@(newSoundID)];
```

Comment:
Much closer, but I think we're going to need to wrap the `SystemSoundID` in a thin class that calls `AudioServicesDisposeSystemSoundID` when it is deallocated.

Reply:
Oh thanks, that was dumb. 

As you've recommended, created a private helper class to manage the resource. PTAL @charlesmchen